experiments/storm/src/main/scala/io/gearpump/experiments/storm/util/Grouper.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.gearpump.experiments.storm.util
+
+import java.util.{List => JList}
+
+import backtype.storm.generated.GlobalStreamId
+import backtype.storm.grouping.CustomStreamGrouping
+import backtype.storm.task.TopologyContext
+import backtype.storm.tuple.Fields
+import clojure.lang.RT
+
+import scala.util.Random
+import scala.collection.JavaConversions._
+
+case class GroupingContext(topologyContext: TopologyContext, globalStreamId: GlobalStreamId, sourceTasks: JList[Integer],
+      outFields: Fields, taskToPartition: Map[Integer, Int])
+
+trait Grouper {
+  def getPartition(taskId: Int, values: JList[AnyRef]): Int
+  def getPartitions(taskId: Int, values: JList[AnyRef]): List[Int] = {
+    List(getPartition(taskId, values))
+  }
+}
+
+class GlobalGrouper extends Grouper {
+  override def getPartition(taskId: Int, values: JList[AnyRef]): Int = 0
+}
+
+class NoneGrouper(numTasks: Int) extends Grouper {
+  val random = new Random
+  lazy val mod = RT.`var`("clojure.core", "mod")
```

Comment:
Check the doc of Non-grouping at 
```
https://storm.apache.org/documentation/Concepts.html
```

I think it is OK to use ShuffleGrouping to replace it.